python/ray/data/dataset.py

```diff
@@ -697,16 +698,21 @@ def _map_batches_without_batch_size_validation(
     def add_column(
         self,
         col: str,
-        fn: Callable[["pandas.DataFrame"], "pandas.Series"],
+        fn: Callable[
+            [DataBatch],
+            Union["pyarrow.Array", "pandas.Series", Dict[str, "np.ndarray"]],
```

Comment:
```suggestion
            Union["pyarrow.ChunkedArray", "pyarrow.Array", "pandas.Series", "np.ndarray"],
```

Reply:
Let'a alias this union as `DataBatchColumn` as well (and add a comment elaborating what it is